### ISSUE TITLE
Segmentation fault in lax._select_and_scatter_add

### ISSUE THREAD

**Issue by bchetioui** [google/jax#4690]
While attempting to add testing code for the JAX2TF conversion of `select_and_scatter_add`, I ran into the following case which triggered a Segmentation Fault:

```python
import jax
from jax import lax
import numpy as np

operand = np.ones((2, 4, 6), dtype=np.float32)
source = np.ones((4, 8, 12), dtype=np.float32)

kwargs = { "select_prim": lax.ge_p
         , "window_dimensions": (2, 2, 2)
         , "window_strides": (1, 1, 1)
         , "padding": ((1, 2), (2, 3), (3, 4))
         }
       
lax._select_and_scatter_add(source, operand, **kwargs)
```

Stack trace:
```
  File "/home/bchetioui/jax/jax/interpreters/xla.py", line 1035 in _value
  File "/home/bchetioui/jax/jax/interpreters/xla.py", line 1007 in __init__
  File "/home/bchetioui/jax/jax/interpreters/xla.py", line 351 in _execute_compiled_primitive
  File "/home/bchetioui/jax/jax/interpreters/xla.py", line 234 in apply_primitive
  File "/home/bchetioui/jax/jax/core.py", line 573 in process_primitive
  File "/home/bchetioui/jax/jax/core.py", line 263 in bind
  File "/home/bchetioui/jax/jax/_src/lax/lax.py", line 1292 in _select_and_scatter_add
  ...
```

Assuming this must come from the implementation of `jax.lib.xla_client.ops.SelectAndScatterWithGeneralPadding`, but I have not yet taken the time to look into it further.

**Comment by hawkinsp** [comment]
I added this to internal bug b/161704903. I think non-standard paddings aren't handled correctly on the CPU backend.

**Comment by hawkinsp** [comment]
Fixed by #6010
